### DISCUSSION TITLE
Installation on Synology with mariadb - What worked for me

### DISCUSSION THREAD

**Discussion by Chacsam** [photoprism/photoprism#3141] [General]
Hi all,

for the ones looking for it, I have been able to easily install photoprism on a synology nas with mariaDB, BUT not via DSM.
The MariaDB installed in DSM (7.1) is version 10.3, not compatible with photoprism.
So my trick was:
- Install MariaDB as a Docker container (I have followed [this tutorial](https://mariushosting.com/how-to-install-mariadb-on-your-synology-nas/))
- Install phpmydamin as a Docker container as well if you want recent version (I have followed [this tutorial](https://mariushosting.com/how-to-install-phpmyadmin-on-your-synology-nas/))
- Go to phpmyadmin, to create a new 'photoprism' database in Mariadb with access to a photoprism user, basically

> - `create database photoprism;`
> - `grant all privileges on photoprism.* TO 'photoprism'@'%' identified by 'YOURPASSWORD';`
> - `flush privileges;`

Follow the [Photoprism on Synology instructions](https://docs.photoprism.app/getting-started/nas/synology/) where you need te replace the following information:
PHOTOPRISM_DATABASE_DRIVER :  mysql
PHOTOPRISM_DATABASE_SERVER :  xxx.xxx.xxx.xxx:3306   (where xxx.xxx.xxx.xxx is your Synology NAS)
PHOTOPRISM_DATABASE_PASSWORD :  YOURPASSWORD  (as defined above)

That did the trick for me. Hopefully it works for you as well

**Comment by graciousgrey** [comment]
Thanks for sharing!

That might be a good fit for https://github.com/photoprism/photoprism-contrib :)

**Comment by psiberfunk** [comment]
It looks like DSM 7.2 will support a new docker interface , including supporting Docker Compose, which should make all this better, I assume? https://prerelease.synology.com/en-global/download/dsm72_beta?builtin_name=synology_geminilake_920%2B
